lib/process-adapter.js

```diff
@@ -0,0 +1,110 @@
+'use strict';
+var fs = require('fs');
+var path = require('path');
+var chalk = require('chalk');
+var sourceMapSupport = require('source-map-support');
+var installPrecompiler = require('require-precompiled');
+
+var debug = require('debug')('ava');
+
+// check if the test is being run without AVA cli
+var isForked = typeof process.send === 'function';
+
+if (!isForked) {
+	var fp = path.relative('.', process.argv[1]);
+
+	console.log();
+	console.error('Test files must be run with the AVA CLI:\n\n    ' + chalk.grey.dim('$') + ' ' + chalk.cyan('ava ' + fp) + '\n');
+
+	process.exit(1); // eslint-disable-line xo/no-process-exit
+}
+
+module.exports.send = function (name, data) {
+	process.send({
+		name: 'ava-' + name,
+		data: data,
+		ava: true
+	});
+};
+
+module.exports.on = process.on.bind(process);
+module.exports.emit = process.emit.bind(process);
+module.exports.exit = process.exit.bind(process);
+module.exports.env = process.env;
+
+var opts = module.exports.opts = JSON.parse(process.argv[2]);
+
+// Fake TTY support
+if (opts.tty) {
+	process.stdout.isTTY = true;
+	process.stdout.columns = opts.tty.columns || 80;
+	process.stdout.rows = opts.tty.rows;
+
+	var tty = require('tty');
+	var isatty = tty.isatty;
+
+	tty.isatty = function (fd) {
+		if (fd === 1 || fd === process.stdout) {
+			return true;
+		}
+
+		return isatty(fd);
+	};
+}
+
+if (debug.enabled) {
+	// Forward the `time-require` `--sorted` flag.
+	// Intended for internal optimization tests only.
+	if (opts._sorted) {
+		process.argv.push('--sorted');
+	}
+
+	require('time-require');
+}
+
+var sourceMapCache = Object.create(null);
+var cacheDir = opts.cacheDir;
+
+module.exports.installSourceMapSupport = function () {
+	sourceMapSupport.install({
+		environment: 'node',
+		handleUncaughtExceptions: false,
+		retrieveSourceMap: function (source) {
+			if (sourceMapCache[source]) {
+				return {
+					url: source,
+					map: fs.readFileSync(sourceMapCache[source], 'utf8')
+				};
+			}
+		}
+	});
+};
+
+module.exports.installPrecompilerHook = function () {
+	installPrecompiler(function (filename) {
+		var precompiled = opts.precompiled[filename];
+
+		if (precompiled) {
+			sourceMapCache[filename] = path.join(cacheDir, precompiled + '.js.map');
+			return fs.readFileSync(path.join(cacheDir, precompiled + '.js'), 'utf8');
+		}
+
+		return null;
+	});
+};
+
+module.exports.installDependencyTracking = function (dependencies, testPath) {
+	Object.keys(require.extensions).forEach(function (ext) {
+		var wrappedHandler = require.extensions[ext];
+		require.extensions[ext] = function (module, filename) {
+			if (filename !== testPath) {
+				dependencies.push(filename);
+			}
+			wrappedHandler(module, filename);
+		};
+	});
+};
+
+module.exports.startTests = function () {
+	console.log('Do not use processAdapter.startTests() in Node. Only browsers');
```

Comment:
Why? What is it for? I can't see it being called here at all.

`Node` => `Node.js`